### PR TITLE
✨ ipam display count of address claims bound in condition

### LINKED ISSUE #1689: IPAM status conditions enhancements
/kind feature

**Describe the solution you'd like**

The `IPAddressClaimed` status condition for IPAM IPAddressClaim and IPAddress reconciliation shows only the first encountered invalid IPAddress, if one is deemed to be invalid. Other IPAddressClaim/IPAddress pairs may be successfully reconciled, but the messages do not show this information.

The condition messages should be enhanced to show the number of pending/total addresses.

This controller was cited as an example of how it may look: https://github.com/kubernetes-sigs/cluster-api/blob/bda7e5cca7b57767437f3627e8cf2d3d389e173f/internal/controllers/machineset/machineset_controller.go

This is a follow on to this PR: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1666

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-vsphere version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by adobley**
**What this PR does / why we need it**:
while there are unbound ip address claims the condition will show the count of bound to total claims, giving some feedback on how much work is left

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1689 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
when using AddressesFromPools the vSphereMachine status will reflect how many claims are awaiting addresses
```

### CODE REVIEW

File: pkg/services/govmomi/service.go
Comment (srm09) [inline pkg/services/govmomi/service.go:451]: even with the error we should check for the state of the other IPAddressClaim objects and aggregate any errors at the end of the for loop, construct an error and return.

**Review by srm09 — CHANGES_REQUESTED** [review]
some linter fixes to be made as well.
/ok-to-test

File: pkg/services/govmomi/service.go
Comment (tylerschultz) [reply pkg/services/govmomi/service.go:451]: While I agree with the spirit of this suggestion, I'm not sure I agree with this one specifically.

The prior step in this reconciliation loop does a find or create for the claims. 

In the current step of the loop, the controller is waiting for the claim to have an address bound. The case where the claim is not found is handled. The only reason this if condition would be hit is if there is an api error. Continuing on and attempting to look up other claims will likely fail too. Gathering up more api errors seemingly provides diminished value.

Unless you think it is important, we'll skip on this suggestion, and instead look for other places where we can apply the suggested strategy. 

WDYT?

**Comment by tylerschultz** [comment]
FYI - We think we've covered the feedback, but we haven't run this code and tried it out. We'll let you know when we think we're ready for another round of review.

**Comment by tylerschultz** [comment]
Update: we've deployed the changes and we are happy with what we're seeing. 
- The condition status shows "0 of 1..." in a nice way when the vm is waiting for IPs.
- The status messages format nicely when multiple IPs cause errors in the VSphereVM yaml.

One area of possible concern would be that these changes include newlines when multiple errors are logged. 

This PR should be good for another round of review.

**Comment by adobley** [comment]
We were seeing build errors fetching a version of vm-operator. We bumped that in another PR (#1732) and have rebased on that branch for now.

**Comment by adobley** [comment]
Due to #1733 we have rebased onto #1734 until that gets merged so that the tests can run.

**Comment by adobley** [comment]
> Due to #1733 we have rebased onto #1734 until that gets merged so that the tests can run.

Rebased back onto master now that #1734 is fixed. This should be good to go once everything passes.

File: pkg/services/govmomi/service.go
Comment (adobley) [reply pkg/services/govmomi/service.go]: @srm09

**Comment by adobley** [comment]
Rebased as it had been a few. I'll keep my eye on this but if everything passes we should be good to go again.

**Pull request merged**
